test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/StrictCheckSpanProcessor.java

```diff
@@ -24,7 +24,7 @@ public class StrictCheckSpanProcessor implements SpanProcessor {
      */
     public StrictCheckSpanProcessor() {}
 
-    private static Map<String, MockSpanData> spanMap = new ConcurrentHashMap<>();
+    private Map<String, MockSpanData> spanMap = new ConcurrentHashMap<>();
```

Comment:
Please revert, `static` was intentional

Reply:
Shall we do one thing, We can expose a ensureSpanStrictCheck method from the MockTelemetry and do validate once the cluster/node is closed from the Test. This way we should make sure that all the nodes are closed and then validate.

Reply:
Yes, we would need essentially something like leak detector , it should run after class very likely, the OpenSearchIntegrTestCase has a hook for it.